core/lib/dal/src/blocks_dal.rs

```diff
@@ -883,6 +884,118 @@ impl BlocksDal<'_, '_> {
         Ok(())
     }
```

Comment:
Meta comment - I am really getting worried about having more and more complexity in the 'dal' methods, without any unittest coverage.